test/safe.ts

```diff
@@ -40,6 +42,35 @@ export class GnosisSafeManager {
     );
   }
 
+  public static async initDeterministic(deployer: Signer): Promise<GnosisSafeManager> {
```

Comment:
Unfortunately the state is not cleared every time we start a new unrelated test, so calling this function more than once will be an issue in the future. This is something that can't be tackled now, just for you to know that you can't rely on this in the future without some extra checks that the deployment destination address won't exist.